src/Fmt_ast.ml

```diff
@@ -132,18 +132,27 @@ let rec sugar_arrow_typ c ({ast= typ} as xtyp) =
   | Ptyp_arrow (l, t1, t2) ->
       Cmts.relocate c.cmts ~src:ptyp_loc ~before:t1.ptyp_loc
         ~after:t2.ptyp_loc ;
-      (l, sub_typ ~ctx t1) :: sugar_arrow_typ c (sub_typ ~ctx t2)
+      let rest =
+        match t2.ptyp_attributes with
+        | [] -> sugar_arrow_typ c (sub_typ ~ctx t2)
```

Comment:
Would it make sense to move the check for empty attributes into `sugar_` instead?

Reply:
I'm not sure what you mean by this. I think we should rewrite theses functions differently to make it clear that they should return a list of size at least 2

Reply:
I was thinking just for `sugar_` to check if the attributes are empty and if not, don't "sugar" and instead return the arg unchanged. That seems to be a common pattern in these changes: either the attributes are empty and the arg is sugared, or the attributes are non-empty and the arg is left unchanged. But maybe the size condition is better.